lib/trino-orc/src/main/java/io/trino/orc/TupleDomainOrcPredicate.java

```diff
@@ -145,7 +147,7 @@ private boolean columnOverlaps(Domain predicateDomain, long numberOfRows, Column
     private static Optional<Collection<Object>> extractDiscreteValues(ValueSet valueSet)
     {
         if (!valueSet.isDiscreteSet()) {
-            return Optional.empty();
+            return tryExpandRanges(valueSet, DEFAULT_COMPACTION_THRESHOLD);
```

Comment:
We need to have tests in TestOrcBloomFilters for range predicates which are under and above the threshold